spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/DeployerSocketUtils.java

```diff
@@ -31,266 +30,148 @@
  * Simple utility methods for working with network sockets &mdash; for example, for
  * finding available ports on {@code localhost}.
  *
- * <p>
- * Within this class, a TCP port refers to a port for a {@link ServerSocket}; whereas, a
- * UDP port refers to a port for a {@link DatagramSocket}.
- *
- * <p>
+ * This is a replacement for SocketUtils.   SocketUtils was introduced in Spring Framework 4.0,
+ * primarily to assist in writing integration tests which start an external server on an available random port.
+ * However, these utilities make no guarantee about the subsequent availability
+ * of a given port and are therefore unreliable. Instead of using SocketUtils to
+ * find an available local port for a server, it is recommended that you rely on a
+ * server's ability to start on a random port that it selects or is assigned by the operating system.
+ * To interact with that server, you should query the server for the port it is currently using.
```

Comment:
If we immediately deprecate this it will also nag us into the reminder that we need to replace it. Wdyt?